src/main/java/pl/project13/maven/git/GitCommitIdMojo.java

```diff
@@ -70,6 +71,12 @@
   @Parameter(property = "session", required = true, readonly = true)
   MavenSession session;
 
+  /**
+   * The Maven settings.
+   */
+  @Parameter(property = "settings", required = true, readonly = true)
+  Settings settings;
```

Comment:
This new settings sadly makes the [tests fail](https://travis-ci.org/git-commit-id/maven-git-commit-id-plugin/builds/559300144).

Inside the [GitIntegrationTest](https://github.com/git-commit-id/maven-git-commit-id-plugin/blob/fc4f5e8e059d40301a8096f93abd2c687cae7b76/src/test/java/pl/project13/maven/git/GitIntegrationTest.java#L118) you also would need to mock them:
```java
  private static Settings mockSettings() {
    Settings settings = mock(Settings.class);
    when(settings.isOffline()).thenReturn(false);
    return settings;
  }
```